### PR TITLE
feat(Input): support `style` prop

### PR DESCRIPTION
**Pull request by argshook**
* remove explicit prop for `width`
* support inline style from `style` prop instead

### CODE REVIEW

File: packages/wix-ui-core/src/components/Input/Input.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ export interface InputProps {
   value?: string;
   id?: string;
   name?: string;
-  width?: string;
+  style?: object;
```
Comment (alisey) [inline packages/wix-ui-core/src/components/Input/Input.tsx]: Should be `React.CSSProperties`

File: packages/wix-ui-core/src/components/Input/Input.spec.tsx [hunk 2]
```diff
@@ -95,7 +95,7 @@ describe('Input', () => {
       const styles = {
         width: '100px',
         background: 'blue',
-        position: 'relative',
+        position: 'relative' as 'relative',
```
Comment (alisey) [inline packages/wix-ui-core/src/components/Input/Input.spec.tsx]: What happened here, does TS complain about something?

Comment (argshook) [reply packages/wix-ui-core/src/components/Input/Input.spec.tsx]: yes, it' throws up

```
Type '{ style: { width: string; background: string; position: string; display: string; }; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Input> & Readonly<{ children?: ReactNode; }> & Rea...'.
  Type '{ style: { width: string; background: string; position: string; display: string; }; }' is not assignable to type 'Readonly<InputProps>'.
    Types of property 'style' are incompatible.
      Type '{ width: string; background: string; position: string; display: string; }' is not assignable to type 'CSSProperties'.
        Types of property 'position' are incompatible.
          Type 'string' is not assignable to type 'PositionProperty'.
```

related https://github.com/Microsoft/TypeScript/issues/11465

Comment (alisey) [reply packages/wix-ui-core/src/components/Input/Input.spec.tsx]: Maybe we could simply remove 'relative' from here in the meanwhile.

**Pull request merged**